src/nodes.coffee

```diff
@@ -2566,7 +2572,11 @@ exports.ObjectProperty = class ObjectProperty extends Base
     keyAst = @key.ast o, LEVEL_LIST
 
     return
-      key: keyAst
+      key:
+        if keyAst?.declaration
```

Comment:
Avoid having a "double declaration" (on both the `key` and `value`) on left-hand-side shorthand object properties eg
```
{a} = ...
```